qiskit_ionq/helpers.py

```diff
@@ -474,6 +477,73 @@ def default(self, o):
         return "unknown"
 
 
+
+def resolve_credentials(token: str = None, url: str = None):
+    """Resolve credentials for use in IonQ API calls.
+
+    If the provided ``token`` and ``url`` are both ``None``, then these values
+    are loaded from the ``IONQ_API_TOKEN`` and ``IONQ_API_URL``
+    environment variables, respectively.
+
+    If no url is discovered, then ``https://api.ionq.co/v0.3`` is used.
+
+    Args:
+        token (str): IonQ API access token.
+        url (str, optional): IonQ API url. Defaults to ``None``.
+
+    Returns:
+        dict[str]: A dict with "token" and "url" keys, for use by a client.
+    """
+    env_token = (
+        dotenv_values().get("QISKIT_IONQ_API_TOKEN")  # first check for dotenv values
+        or dotenv_values().get("IONQ_API_KEY")
+        or dotenv_values().get("IONQ_API_TOKEN")
+        or os.getenv("QISKIT_IONQ_API_TOKEN")  # then check for global env values
+        or os.getenv("IONQ_API_KEY")
+        or os.getenv("IONQ_API_TOKEN")
+    )
+    env_url = (
+        dotenv_values().get("QISKIT_IONQ_API_URL")
+        or dotenv_values().get("IONQ_API_URL")
+        or os.getenv("QISKIT_IONQ_API_URL")
+        or os.getenv("IONQ_API_URL")
+    )
+    return {
+        "token": token or env_token,
+        "url": url or env_url or "https://api.ionq.co/v0.3",
+    }
+
+
+
+def get_n_qubits(backend: str) -> int:
+    """Get the number of qubits for a given backend.
+
+    Args:
+        backend (str): The name of the backend.
+
+    Returns:
+        int: The number of qubits for the backend.
+    """
+    url, token = resolve_credentials().values()
+    try:
+        return requests.get(
+            url=f"{url}/characterizations/backends/{backend}/current",
+            headers={"Authorization": f"apiKey {token}"},
+            timeout=5,
+        ).json()["qubits"]
+    except Exception:  # pylint: disable=broad-except
+        if backend == "ionq_qpu.harmony":
+            return 11
+        elif backend == "ionq_qpu.aria-1":
+            return 25
+        elif backend == "ionq_qpu.aria-2":
+            return 25
+        elif backend == "ionq_qpu.forte-1":
+            return 36
+        else:
+            return 36
```

Comment:
I would have a hard time understanding what this code was doing, since it falls back during an exception but doesn't' log it or what value it's deciding. Could you add logs here? (or alternatively, fall back to Infinity or something >> 36?)

Reply:
we can't do inifinity because the transpiler freezes (i guess it tries to create a graph of length #qubits lol